src/components/engagement/engagement.service.ts

```diff
@@ -258,35 +254,24 @@ export class EngagementService {
   ): Promise<Engagement> {
     const isLanguageEngagement = dto.__typename === 'LanguageEngagement';
 
-    const securedProperties = await this.authorizationService.secureProperties(
-      isLanguageEngagement ? LanguageEngagement : InternshipEngagement,
-      dto,
-      session,
-    );
-
     const canDelete =
       dto.status !== EngagementStatus.InDevelopment &&
       !session.roles.includes(`global:Administrator`)
         ? false
         : await this.repo.checkDeletePermission(dto.id, session);
     if (isLanguageEngagement) {
-      // help TS understand that the secured props are for a LanguageEngagement
-      const secured = securedProperties as SecuredResource<
-        typeof LanguageEngagement,
-        false
-      >;
-
       return {
         ...(dto as UnsecuredDto<LanguageEngagement>),
-        ...secured,
+        ...this.privileges
+          .for(session, LanguageEngagement)
+          .secure(dto as UnsecuredDto<LanguageEngagement>),
         canDelete,
```

Comment:
@CarsonF - I know we talked about the newer privileges having everything we need (including delete permissions) but this feels a little bit different? Can this canDelete still be inferred from the privileges here making the entire `canDelete` variable removable?